### DISCUSSION TITLE
Is it possible to push encoded url params?

### DISCUSSION THREAD

**Discussion by astrodomas** [47ng/nuqs#670] [General] [answered]
Hello,
When I apply a search parameter value it is not encoded `URLSearchParams` encoded. Is it possible to do so in nuqs?

**Comment by franky47** [comment]
Do you have a concrete example where encoding is necessary (ie: it breaks URL parsers without) ?

Not all characters are encoded, see #372 for reference.

**Reply by astrodomas** [reply]
This is what I was refering to. Thank's now it makes sense. However, one question is, how do you compose `useSearchParams` and `nuqs` encoding? As `useSearchParams` query string will differ against `nuqs`. Do I need to write some sort of wrapper in respect to `useSearchParams`?

**Reply by franky47** [reply]
Any encoding shown in the URL should be decoded when accessing URLSearchParams, whether directly via `new URLSearchParams(location.search)`, via `useSearchParams`, or via nuqs. So there should not be a need for a wrapper.

Do you have an example where this is not the case?

**Reply by astrodomas** [reply]
I'll try to provide an example, most likely not today. Thanks!

**Reply by franky47** [reply]
Hey @astrodomas, do you have a follow-up on this?

**Reply by astrodomas** [reply, marked as answer]
Hi, no I think i'll open a new thread once I get into refactoring my project to nuqs again. Sorry for the wait...
